openedx/features/course_duration_limits/models.py

```diff
@@ -75,6 +76,17 @@ def enabled_for_enrollment(cls, enrollment=None, user=None, course_key=None):
         if enrollment is None:
             enrollment = CourseEnrollment.get_enrollment(user, course_key)
 
+        # if the user is has a role of staff, instructor or beta tester their access should not expire
+        if user is None and enrollment is not None:
+            user = enrollment.user
+        if user:
+            staff_role = CourseStaffRole(course_key).has_user(user)
+            instructor_role = CourseInstructorRole(course_key).has_user(user)
+            beta_tester_role = CourseBetaTesterRole(course_key).has_user(user)
```

Comment:
I'm a bit late to the party, but we should probably make this change in `content_type_gating.models` as well, if that's where we've decided to put those checks. I was originally planning to keep the code for actually checking access and partition grouping in their respective files.

I guess my reasoning for moving holdback into is-enabled is that it felt like holdback was a global rule, but particular roles felt more like business rules, and thus felt more appropriate for `access.py` or `partitions.py` and `field_data.py`.

Reply:
I guess I had the wrong interpretation of your suggestion, we could certainly move these things back, I feel pretty flexible regarding this.  
I guess my reasoning was that if we had most of the code to determine if the feature would be enabled in one place, that may simplify things down the road.  
but there may be good reasons for not doing that

Reply:
not saying you're wrong, but could you help me understand this perspective?  
I guess my assumption was that since we would be disabling all parts of the feature for various staff roles it would be better to put the code in one place.  
If we repeat it in many places that would create more opportunities for bugs if we need to change the condition everywhere and miss a spot, or if we forgot to add the check to one of the many flex scope features we will be adding.
It sounds to me like the business rules are that the whole feature is disabled for staff, so what is the benefit of checking in each use case rather than at the feature level?  
again, not saying you're wrong, just want to better understand the argument